report_aeroo/models/report.py

```diff
@@ -274,73 +278,72 @@ def load_from_file(self, path, key):
                     zimp = zipimport.zipimporter(mod_path+os.path.sep+path.split(os.path.sep)[0]+'.zip')
                     return zimp.load_module(path.split(os.path.sep)[0]).parser.Parser
         except SyntaxError as e:
-            raise except_orm(_('Syntax Error !'), e)
+            raise UserError(_('Syntax Error !'), e)
         except Exception as e:
             _logger.error('Error loading report parser: %s'+(filepath and ' "%s"' % filepath or ''), e)
             return None
-    
+
     @api.model
     def create(self, vals):
-        if vals.get('report_type') != 'aeroo':
-            res_id = super(report_aeroo, self).create(vals)
-            return res_id
-        
-        if 'report_type' in vals and vals.get('report_type') == 'aeroo':
-            parser = models.AbstractModel
-            ir_model = self.env['ir.model']
-            model_data = {'model': vals['report_name'],
-                          'name': vals['name'],
-                          'parser_def': vals['parser_def'],
-                         }
-            if vals['parser_state']=='loc' and vals['parser_loc']:
-                parser=self.load_from_file(vals['parser_loc'], vals['name'].lower().replace(' ','_')) or parser #TODO Revise
-            elif vals['parser_state']=='def' and vals['parser_def']:
-                parser = ir_model._custom_aeroo_parser(model_data)
-            elif vals['parser_state']=='default':
-                parser = ir_model._default_aeroo_parser(model_data)
-            model = ir_model._get(vals.get('model'))
-            res_id = super(report_aeroo, self).create(vals)
-            parser._build_model(self.pool, self.env.cr)
-            return res_id
-            
-    @api.one
-    def write(rec, vals):
-        orec = rec.read()[0]
-        if vals.get('report_type') and orec['report_type'] != vals['report_type']:
-            raise UserError(_("Changing report type not allowed!"))
-        
+        rec = super(ReportAeroo, self).create(vals)
```

Comment:
With this we use new odoo orm/api funcionalities and it also fixes that if you where creating a report without sending "parser_state" field, the default value "default" was not being considered